### PR TITLE
Update notebook GLM-negative-binomial-regression to pymc v5 #573

### PR DESCRIPTION
**Pull request by wesleyboelrijk**
# Description
Issue
#573

The notebook GLM-negative-binomial-regression was still using PyMC v4. I have updated it to PyMC v5.
Besides that, there is a dependency on seaborn, which was not listed in requirements-docs.txt. I have added this.
Lastly, seaborn gave a bunch of FutureWarnings upon plotting in the notebook. I added a warning suppression for those FutureWarnings.

Screenshot 2023-09-16 at 15 59 13

+ [x] Notebook follows style guide https://docs.pymc.io/en/latest/contributing/jupyter_style.html
+ [x] PR description contains a link to the relevant issue:
  * a tracker one for existing notebooks (tracker issues have the "tracker id" label)
  * or a proposal one for new notebooks
+ [x] Check the notebook is not excluded from any pre-commit check: https://github.com/pymc-devs/pymc-examples/blob/main/.pre-commit-config.yaml

### Helpful links
* https://github.com/pymc-devs/pymc-examples/blob/main/CONTRIBUTING.md

----
:books: Documentation preview :books:: https://pymc-examples--575.org.readthedocs.build/en/575/

### CODE REVIEW

File: examples/generalized_linear_models/GLM-negative-binomial-regression.ipynb
Comment (drbenvincent) [inline examples/generalized_linear_models/GLM-negative-binomial-regression.ipynb:1]: The author metadata in the first notebook cell should remain unchanged as this refers to the original authors. I believe this is only changed after major and substantive changes (if at all).

File: examples/case_studies/nyc_bym.ipynb
Comment (drbenvincent) [inline examples/case_studies/nyc_bym.ipynb:1]: We've got a minor edit to this unrelated file that seems to have snuck in by accident. Can we revert this change?

File: examples/howto/model_builder.ipynb
Comment (drbenvincent) [inline examples/howto/model_builder.ipynb:1]: We've got a minor edit to this unrelated file that seems to have snuck in by accident. Can we revert this change?

File: requirements-docs.txt
Comment (drbenvincent) [inline requirements-docs.txt:1]: Good... I'm just noticing that these imports are out of alphabetical order. Would you mind just ordering them please?

**Review by drbenvincent — CHANGES_REQUESTED** [review]
Hi @wesleyboelrijk. Thanks for the contribution! Looks great, but I just left a few minor change requests.

File: examples/generalized_linear_models/GLM-negative-binomial-regression.myst.md [hunk 1]
```diff
@@ -30,10 +32,11 @@ from scipy import stats
 ```
 
 ```{code-cell} ipython3
+warnings.simplefilter(action="ignore", category=FutureWarning)
```
Comment (OriolAbril) [inline examples/generalized_linear_models/GLM-negative-binomial-regression.myst.md:35]: It might be possible to avoid ignorning warnings by using the just released seaborn https://github.com/mwaskom/seaborn/releases/tag/v0.13.0rc0.

File: examples/generalized_linear_models/GLM-negative-binomial-regression.ipynb
Comment (OriolAbril) [reply examples/generalized_linear_models/GLM-negative-binomial-regression.ipynb:1]: Yes, given there were no major changes to rerun with v5, you should list yourself at the authors section but not here in the metadata. Ref: https://www.pymc.io/projects/docs/en/latest/contributing/jupyter_style.html#authorship-and-attribution

Comment (wesleyboelrijk) [reply examples/generalized_linear_models/GLM-negative-binomial-regression.ipynb:1]: Thanks for the feedback, I was not aware of this part of the guide yet. I have updated the notebook accordingly!

File: examples/case_studies/nyc_bym.ipynb
Comment (wesleyboelrijk) [reply examples/case_studies/nyc_bym.ipynb:1]: They snuck in through the pre-commit. I removed them from the PR manually.

File: examples/generalized_linear_models/GLM-negative-binomial-regression.myst.md (hunk 1 above)
Comment (wesleyboelrijk) [reply examples/generalized_linear_models/GLM-negative-binomial-regression.myst.md]: That's right! Thanks for the heads up. I have removed the warning suppression from the notebook.

**Comment by wesleyboelrijk** [comment]
> Hi @wesleyboelrijk. Thanks for the contribution! Looks great, but I just left a few minor change requests.

Thanks @drbenvincent and @OriolAbril for the feedback! I made the requested changes. Let me know if any other changes are required :)

File: requirements-docs.txt [hunk 2]
```diff
@@ -1,12 +1,13 @@
+ablog<0.11
+matplotlib
 myst-nb
-sphinx>=5
 pymc-sphinx-theme==0.1
-sphinx-design
+seaborn
```
Comment (OriolAbril) [inline requirements-docs.txt:5]: seaborn should not be added here. Sorry I missed that in the previous round.

These are the requirements to **build** the website (you'll see not even pymc is here). The requirements to write notebooks are the `-write` ones. We could add seaborn there, but I would argue against this. Barely anybody knows about this, they see the notebook on the website and try to run it, often not even finding this repo at all. As seaborn is not a pymc dependency, I think it should be added using https://www.pymc.io/projects/docs/en/latest/contributing/jupyter_style.html#extra-dependencies

**Review by OriolAbril — CHANGES_REQUESTED** [review]
missed an important part in the previous review

File: requirements-docs.txt (hunk 2 above)
Comment (wesleyboelrijk) [reply requirements-docs.txt]: That's a good point! Seaborn has been removed from the `requirements-docs.txt` and added to the notebook-specific metadata.

**Review by OriolAbril — APPROVED** [review]

**Review by OriolAbril — APPROVED** [review]
